sti-main/src/main/java/uk/ac/shef/dcs/sti/core/algorithm/tmp/TMPOdalicInterpreter.java

```diff
@@ -71,6 +73,26 @@ public TAnnotation start(Table table, Constraints constraints) throws STIExcepti
           .compute(table, constraints.getSubjectColumnPosition(), ignoreColumnsArray);
       tableAnnotations.setSubjectColumn(subjectColumnScores.get(0).getKey());
       
+      // when the column does not contain Named entity as the most frequent data type,
+      // we will not disambiguate (and so classify) them, except for
+      // the user defined constraints
+      Set<Ambiguity> newAmbiguities = new HashSet<>(constraints.getAmbiguities());
```

Comment:
I do not like that for file which contains 1000 rows and 10 numerical columns, it creates 10*1000 entries. It would be more efficient to store just the user defined constraints (which we store anyway, right) and then just store e.g. the number of the column which is not classified as named entity, but which contains disambiguation, right?  Or what is the advantage to store  such "Ambiguities" for each cell (even though in most cases it will be the same for all cell values within the column classified as not being a named entity)?